pkg/registry/resource/resourceclaimtemplate/strategy_test.go

```diff
@@ -848,3 +892,58 @@ func TestStrategyUpdate(t *testing.T) {
 		})
 	}
 }
+
+func TestFeatureInUse(t *testing.T) {
+	testcases := map[string]struct {
+		obj       *resource.ResourceClaimTemplate
+		inUseFunc func(spec *resource.ResourceClaimSpec) bool
+		expect    bool
```

Comment:
Is the expectation that this might be useful for other functions? If not, then this could be called `TestDRAConsumableCapacityFeatureInUse` and `inUseFunc` can be removed.

But I am wondering: as this tests DRAConsumableCapacityFeatureInUse, shouldn't it be in the `resourceclaimspec` package?

You can copy-and-paste the object construction. Some code duplication can make tests more readable and self-contained.

Reply:
@pohly I think this can be reused for the other inuse check in the other feature as well so that I define inUseFunc. However, I'm not planning to add tests for the other features in this PR. Should I reduce it to focus on ConsumableCapacityFeatureInUse function? For either way, I will move it to the resourceclaimspec package.